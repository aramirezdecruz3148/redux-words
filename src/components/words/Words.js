import React from 'react';
import PropTypes from 'prop-types';

function Words({ words }) {
  const wordsList = words.map((word) => (
    <li key={word}>
      <h3>{word}</h3>
    </li>
  ));
  return (
    <ul>
      {wordsList}
    </ul>
  );
}

Words.propTypes = {
  words: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Words;
