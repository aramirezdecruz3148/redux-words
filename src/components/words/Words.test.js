import React from 'react';
import { shallow } from 'enzyme';
import Words from './Words';

describe('Words component', () => {
  it('renders Words', () => {
    const wrapper = shallow(<Words words={['a', 'string']} />);
    expect(wrapper).toMatchSnapshot();
  });
});
