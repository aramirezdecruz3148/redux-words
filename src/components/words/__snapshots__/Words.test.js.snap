// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Words component renders Words 1`] = `
<ul>
  <li
    key="a"
  >
    <h3>
      a
    </h3>
  </li>
  <li
    key="string"
  >
    <h3>
      string
    </h3>
  </li>
</ul>
`;
